crates/tuono_lib_macros/src/handler.rs

```diff
@@ -1,77 +1,94 @@
-use crate::utils::{
-    crate_application_state_extractor, create_struct_fn_arg, import_main_application_state,
-    params_argument, request_argument,
-};
-
 use proc_macro::TokenStream;
 use quote::quote;
 use syn::punctuated::Punctuated;
+use syn::spanned::Spanned;
 use syn::token::Comma;
-use syn::{FnArg, ItemFn, Pat, parse_macro_input};
+use syn::{parse_macro_input, parse_quote, Expr, FnArg, Ident, ItemFn, Pat, PatIdent, Type};
 
 pub fn handler_core(_args: TokenStream, item: TokenStream) -> TokenStream {
     let item = parse_macro_input!(item as ItemFn);
 
     let fn_name = &item.sig.ident;
 
-    let mut argument_names: Punctuated<Pat, Comma> = Punctuated::new();
-    let mut axum_arguments: Punctuated<FnArg, Comma> = Punctuated::new();
-
-    // Fn Arguments minus the first which always is the request
-    for (i, arg) in item.sig.inputs.iter().enumerate() {
-        if i == 0 {
-            axum_arguments.insert(i, params_argument());
-            continue;
-        }
+    let mut arguments: Punctuated<FnArg, Comma> = Punctuated::new();
+    let mut passed_arguments: Punctuated<Expr, Comma> = Punctuated::new();
 
-        if i == 1 {
-            axum_arguments.insert(1, create_struct_fn_arg())
-        }
+    let mut request_arg_index: Option<usize> = None;
 
+    for (i, arg) in item.sig.inputs.iter().enumerate() {
         if let FnArg::Typed(pat_type) = arg {
-            let index = i - 1;
-            let argument_name = *pat_type.pat.clone();
-            argument_names.insert(index, argument_name.clone());
+            match &*pat_type.pat {
+                Pat::Ident(PatIdent { ident, .. }) => {
```

Comment:
This section could use a second pair of eyes. Here I check each of the parameters to see if it's a request argument and save it off. I check the last segment so if the user specified `axum::extract::Request` it would still work _but_ if they specified `some::other::Request` it would take that as the request argument. I'm not sure of a good way around that or if it matters.

We need to know if there's a request argument and what it's called so that we can pass it into `render_to_string()`. I did look into making this a tower service, but the mechanics of how Handler interacts with a Tower layer needed more time to wrap my head around, and this approach doesn't have the runtime overhead of a layer.

Reply:
Ok, pipelines should be fixed now the e2e tests pass locally for me